flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/IncrementalAggregateReduceFunction.scala

```diff
@@ -52,9 +53,14 @@ class IncrementalAggregateReduceFunction(
     //   and directly merge value1 and value2.
     val accumulatorRow = new Row(intermediateRowArity)
 
-    // copy all fields of value1 into accumulatorRow
-    (0 until intermediateRowArity)
-    .foreach(i => accumulatorRow.setField(i, value1.getField(i)))
+    // copy non agg fields of value2 into accumulatorRow
+    (0 until aggOffset)
+      .foreach(i => accumulatorRow.setField(i, value2.getField(i)))
+
+    // copy agg fields of value1 into accumulatorRow
```

Comment:
IMO, groupWindow's non-agg fields is group fields, and whether copy from value1 or value2 is the same, what do you think?